src/calculateRentalCost.js

```diff
@@ -1,10 +1,23 @@
 /**
  * @param {number} days
- *
  * @return {number}
  */
 function calculateRentalCost(days) {
-  // write code here
+  const DAILY_RENTAL_COST = 40;
+  const DISCOUNT_FOR_3_DAYS = 20;
+  const DISCOUNT_FOR_7_DAYS = 50;
+
+  const totalCost = days * DAILY_RENTAL_COST;
+
+  if (days >= 7) {
+    return totalCost - DISCOUNT_FOR_7_DAYS;
+  }
+
+  if (days >= 3) {
+    return totalCost - DISCOUNT_FOR_3_DAYS;
+  }
```

Comment:
The current discount logic only applies one discount, either for 3 days or 7 days, but not both. According to the task requirements, the logic should be adjusted to ensure that both discounts are applied when applicable. Consider restructuring the conditional statements to correctly apply the discounts based on the number of days.